src/datascience-ui/data-explorer/reactSlickGrid.tsx

```diff
@@ -458,6 +474,11 @@ export class ReactSlickGrid extends React.Component<ISlickGridProps, ISlickGridS
         return null;
     }
 
+    private updateColumns = (_e: Slick.EventData, newColumns: Slick.Column<Slick.SlickData>[]) => {
+        this.state.grid?.setColumns(newColumns);
+        this.state.grid?.render(); // We might be able to skip this rerender?
```

Comment:
Looks like an open comment here. Have you tried it without?

Reply:
Yes and it worked, but the API documentation says

> Column headers will be recreated and all rendered rows will be removed. To re-render the grid (if necessary), call render().

So not really sure what to make of that.